jib-core/src/main/java/com/google/cloud/tools/jib/registry/credentials/DockerCredentialHelper.java

```diff
@@ -88,10 +106,30 @@ public DockerCredentialHelper(String serverUrl, Path credentialHelper) {
   public Credential retrieve()
       throws IOException, CredentialHelperUnhandledServerUrlException,
           CredentialHelperNotFoundException {
+    boolean isWindows =
+        systemProperties.getProperty("os.name").toLowerCase(Locale.ENGLISH).contains("windows");
```

Comment:
Do you think we might compute this value and pass it into the class?  We're already computing it in `DefaultCredentialRetrievers`.

Reply:
So I've looked into the chain, and it goes through

1. `DefaultCredentialRetrievers.asList()`
2. `CredentialRetrieverFactory.dockerCredentialHelper(String)`
3. `CredentialRetrieverFactory.dockerCredentialHelper(Path)`
4. `CredentialRetrieverFactory.retrieveFromDockerCredentialHelper()`
5. `DockerCredentialHelper.create()` (which is `DockerCredentialHelper::new`)

It's a long chain. And I think some in the chain don't look reasonable to require passing `isWindows` in their signature. Moreover, this is relatively simple code and can be done in a self-contained manner, I think this is OK.